File: 24-Template-Method/src/go/coffee.go
package main

import "fmt"

// Coffee fills in the varying steps; it embeds DefaultBeverage for the hook.
type Coffee struct {
	DefaultBeverage
}

func (Coffee) Brew()          { fmt.Println("  Dripping coffee through filter") }
func (Coffee) AddCondiments() { fmt.Println("  Adding sugar and milk") }

File: 24-Template-Method/src/rust/coffee.rs
use crate::beverage::CaffeineBeverage;

/// Fills in the varying steps for coffee.
pub struct Coffee;

impl CaffeineBeverage for Coffee {
    fn brew(&self) {
        println!("  Dripping coffee through filter");
    }

    fn add_condiments(&self) {
        println!("  Adding sugar and milk");
    }
}
